tests/Ontology/TransactionTests.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "HexCoding.h"
+#include "Ontology/Signer.h"
+#include "Ontology/Transaction.h"
+#include "Ontology/ParamsBuilder.h"
+
+#include <unordered_map>
+
+#include <PrivateKey.h>
+
+#include <gtest/gtest.h>
+#include <boost/any.hpp>
+
+using namespace TW;
+using namespace TW::Ontology;
+
+
+TEST(OntologyTransaction, validity) {
+    std::vector<uint8_t> ontContract{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01};
+    auto fromAddress = Address("AeicEjZyiXKgUeSBbYQHxsU1X3V5Buori5");
+    auto toAddress = Address("APniYDGozkhUh8Tk7pe35aah2HGJ4fJfVd");
+    uint64_t amount = 1;
+    std::unordered_map<std::string, boost::any> transferParam{{"from", fromAddress.data}, {"to", toAddress.data}, {"amount", amount}};
+    std::vector<boost::any> args{transferParam};
+    auto invokeCode = ParamsBuilder::buildNativeInvokeCode(ontContract, 0x00, "transfer", args);
+    uint8_t version = 0;
+    uint8_t txType = 0xd1;
+    uint32_t nonce = 1552759011;
+    uint64_t gasPrice = 600;
+    uint64_t gasLimit = 300000;
+    auto tx = Transaction(version, txType, nonce, gasPrice, gasLimit, toAddress.string(), invokeCode);
+    std::string hexTx = "00d1e3388d5c5802000000000000e09304000000000057e9d1a61f9aafa798b6c7fbeae35639681d7df67100c66b14fbacc8214765d457c8e3f2b5a1d3c4981a2e9d2a6a7cc81457e9d1a61f9aafa798b6c7fbeae35639681d7df66a7cc8516a7cc86c51c1087472616e736665721400000000000000000000000000000000000000010068164f6e746f6c6f67792e4e61746976652e496e766f6b650000";
+    EXPECT_EQ(hexTx, hex(tx.serialize()));
```

Comment:
move all expect/assert together or leave a new empty line, easier to read

Reply:
I split them into more short test case, maybe this will make it easier to read. https://github.com/TrustWallet/wallet-core/pull/201/commits/c6e54247fb7d728bbbc9712cedd4ca91bcd4814a